volatility3/plugins/windows/registry/certificates.py

```diff
@@ -63,10 +69,11 @@ def _generator(self) -> Iterator[Tuple[int, Tuple[str, str, str, str]]]:
                             key_hash = key_path[key_path.rindex("\\") + 1:]
 
                             if self.config['dump']:
-                                if not isinstance(certificate_data, interfaces.renderers.BaseAbsentValue):
-                                    with self.open("{} - {} - {}.crt".format(hex(hive.hive_offset), reg_section,
-                                                                            key_hash)) as file_data:
-                                        file_data.write(certificate_data)
+                                self.dump_data(certificate_data, hive.hive_offset, reg_section, key_hash)
+                            else:
+                                vollog.warning("Certificates plugin is no longer support automatically dumped, please use the dump option.")
+                                self.dump_data(certificate_data, hive.hive_offset, reg_section, key_hash)
+                            
```

Comment:
Originally, there was a code that considered side effects because the dump was automatic, but now required options. However, I wondered if it was appropriate because it was not an official plugin and was unlikely to be managed in the future. Do you think it's necessary?

Reply:
Hmmmm, I don't think this is the best way of handling this.  It's a bit strange to say it won't dump the certificates and then dump them anyway, and it will also likely be annoying to people to get warned that you need to turn on a flag, given that's how the plugin (now) works.  I understand the point of informing the user, but I think it'll confuse people in the long run, than the one off surprise of finding out the plugin has changed.

It might help if the output were more significantly changed.  As I say, we don't intend to support this, so this is a purely a suggestion not a requirement for this pull request, but the plugin could try loading the certificate and try extracting the common name?  It can just return an AbsentValue if it can't find read it, but it might be slightly more helpful for users and make the non-dump option more useful?

Reply:
@ikelos Thank you for your opinion! Could you explain in more detail?
I think I don't understand all the meaning of my English skills. :(

Reply:
If you don't have any additional comments on this, you can merge it.
Rather, I think it's an opportunity to focus on what I wrote in other draft PR. :)